x/builder/ante/ante.go

```diff
@@ -12,20 +12,12 @@ import (
 
 var _ sdk.AnteDecorator = BuilderDecorator{}
 
-type (
-	BuilderDecorator struct {
-		builderKeeper keeper.Keeper
-		txDecoder     sdk.TxDecoder
-		txEncoder     sdk.TxEncoder
-		mempool       *mempool.AuctionMempool
-	}
-
-	TxWithTimeoutHeight interface {
-		sdk.Tx
-
-		GetTimeoutHeight() uint64
-	}
-)
+type BuilderDecorator struct {
+	builderKeeper keeper.Keeper
+	txDecoder     sdk.TxDecoder
+	txEncoder     sdk.TxEncoder
+	mempool       *mempool.AuctionMempool
```

Comment:
Instead of having a complete dependency on the concrete mempool implementation, why not instead have the `Config` interface? If the decorator expands to us needing more functionality from the mempool, then we can possibly add that dependency.

https://en.wikipedia.org/wiki/Law_of_Demeter

Reply:
I am actually doing that in this PR: https://github.com/skip-mev/pob/pull/81.